jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamTemplateUtils.kt

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+@file:JvmName("TemplateUtils")
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VirtualFile
+import org.jetbrains.yaml.psi.YAMLPsiElement
+import software.aws.toolkits.jetbrains.services.cloudformation.CloudFormationTemplate
+import software.aws.toolkits.jetbrains.services.cloudformation.LambdaFunction
+import software.aws.toolkits.jetbrains.services.cloudformation.Resource
+import software.aws.toolkits.resources.message
+import java.io.File
+
+private const val SERVERLESS_FUNCTION_TYPE = "AWS::Serverless::Function"
+private const val LAMBDA_FUNCTION_TYPE = "AWS::Lambda::Function"
+
+class SamFunction(private val delegate: Resource) : LambdaFunction(delegate) {
+    override fun codeLocation(): String {
+        return getScalarProperty("CodeUri")
+                ?: throw IllegalStateException("CodeUri property is not set for $logicalName")
+    }
+
+    override fun setCodeLocation(location: String) {
+        setScalaProperty("CodeUri", location)
+    }
+}
+
+fun findSamFunctionsFromTemplate(project: Project, file: File): List<LambdaFunction> {
+    val virtualFile = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file) ?: throw RuntimeException(
+        message("lambda.sam.template_not_found", file)
+    )
+    return findSamFunctionsFromTemplate(project, virtualFile)
+}
+
+fun findSamFunctionsFromTemplate(project: Project, file: VirtualFile): List<LambdaFunction> {
```

Comment:
It really should be a `findLambdaFunctionsFromTemplate` from the implementation as it returns a list of `LambdaFunction`

Reply:
renamed to findFunctionsFromTemplate